Documentation and change logs for pronouns settings (#19632)
microsoftgraph/microsoft-graph-docs: concepts/pronouns-configure-pronouns-availability.md

---

> title: "Enabling or disabling pronouns in Microsoft 365 (preview)"

Reviewer: is this redundant ? others dont have it

description: "Configure availability of pronouns by using REST API pronounsSettings resource."
author: "aymen-ms"
ms.localizationpriority: high
ms.prod: "people"
ms.custom: scenarios:getting-started
---

# Enabling or disabling pronouns settings in Microsoft 365 (preview)

> [!NOTE]
> APIs under the `/beta` version in Microsoft Graph are subject to change. Use of these APIs in production applications is not supported.

> If you are a global administrator, you can enable and disable showing pronouns in your organization. Pronouns are **off** by default. When pronouns are enabled, users can add, update, and delete pronouns in Microsoft apps and services.

Reviewer: does these map to any actions we expose or its a form of speech ?

Author: Just as an intro to the guide

Pronouns settings are exposed through the *pronouns* property in *organizationSettings*.

> ## Confirm your current settings

Reviewer: Looking at docs for itemInsights, we will have also powershell commands when we expose to beta, maybe we can document powershell rather on this page ?

Author: For the cmdlets to be generated we need to have the api published, we can update that once we have the api published

``` http
PATCH https://graph.microsoft.com/beta/organization/{organizationId}/settings/pronouns
Content-Type: application/json

{
  "isEnabledInOrganization": true
}
```
If successful, the response returns a `200 OK` response code and a [**pronounsSettings**](/graph/resources/pronounssettings?view=graph-rest-beta&preserve-view=true) object in the response body.

``` http
HTTP/1.1 200 OK
Content-Type: application/json

{
  "isEnabledInOrganization": true
}
```

## Disabling pronouns in your organization

Use the `patch` operation to make pronouns unavailable in Microsoft apps and services in your organization.

``` http
PATCH https://graph.microsoft.com/beta/organization/{organizationId}/settings/pronouns
Content-Type: application/json

{
  "isEnabledInOrganization": false
}
```

If successful, the response returns a `200 OK` response code and a [**pronounsSettings**](/graph/resources/pronounssettings?view=graph-rest-beta&preserve-view=true) object in the response body.

``` http
HTTP/1.1 200 OK
Content-Type: application/json

{
  "isEnabledInOrganization": false
}
```

## Important

> When you turn off pronouns in your organization, pronouns are immediately hidden in Microsoft apps, but the data can persist for up to 30 days on Microsoft servers. If you choose to enable pronouns back not yet deleted pronouns that were hidden will be visible again. 

Reviewer: i wonder if there should be any commas or similar in this sentence

---

> title: "Enabling or disabling pronouns (preview)"

Reviewer: this file is still missing in TOC if you go to this link https://review.learn.microsoft.com/en-us/graph/api/pronounssettings-update?view=graph-rest-beta&branch=pr-en-us-19632 and then try to expand API Beta Reference -> People and Workplace Intelligence - it wont be there

``` http
PATCH https://graph.microsoft.com/beta/organization/{organizationId}/settings/pronouns
Content-Type: application/json

{
  "isEnabledInOrganization": true
}
```
If successful, the response returns a `200 OK` response code and a [**pronounsSettings**](/graph/resources/pronounssettings?view=graph-rest-beta&preserve-view=true) object in the response body.

``` http
HTTP/1.1 200 OK
Content-Type: application/json

{
  "isEnabledInOrganization": true
}
```

## Disabling pronouns in your organization

Use the `patch` operation to make pronouns unavailable in Microsoft apps and services in your organization.

``` http
PATCH https://graph.microsoft.com/beta/organization/{organizationId}/settings/pronouns
Content-Type: application/json

{
  "isEnabledInOrganization": false
}
```

If successful, the response returns a `200 OK` response code and a [**pronounsSettings**](/graph/resources/pronounssettings?view=graph-rest-beta&preserve-view=true) object in the response body.

``` http
HTTP/1.1 200 OK
Content-Type: application/json

{
  "isEnabledInOrganization": false
}
```

## Important

> When pronouns are turned off in your organization, pronouns will be hidden in Microsoft apps and services. However, the data may persist on Microsoft servers for up to 30 days. If you decide to re-enable pronouns within that period, any previously hidden pronouns that have not yet been deleted will became visible again.

Reviewer: we should add explicitly that they will be deleted as well.

---

> title: "Customize pronouns availability (preview) (preview)"

Reviewer 2: This is more of a question than an observation. Is there a particular reason why the word "preview" appears twice?

> description: "Configure the availability of pronouns by using the REST API pronounsSettings resource."

Reviewer 3: Line 2:
Please specify the same value as the topic title.
Per my comment below, I'm suggesting the topic title to be:
"Manage the display of pronouns in profile cards using the Microsoft Graph API"

Reviewer 3: Since I'm going to make a commit in this PR for this topic, I will include the title change to the commit.

Please let me know if you see an issue with the retitling.

Reviewer 3: Per Anton's email on 1/18 that displaying pronouns is one result of toggling on, I'm scoping down to a specific section in the concept topic where I list the exact effects of "enabling" pronouns, and for other places, like the topic and TOC titles, I try not to tie them to "display" anymore.
FYI I'm changing the topic title to "Manage pronouns settings for an organization using the Microsoft Graph API".

> description: "Configure availability of pronouns by using the REST API pronounsSettings resource."

Reviewer 2: ```suggestion
description: "Configure the availability of pronouns by using the REST API pronounsSettings resource."
```

author: "aymen-ms"
ms.localizationpriority: high
ms.prod: "people"
ms.custom: scenarios:getting-started
---

# Manage access to pronouns using APIs in Microsoft Graph (preview)

> 

Reviewer 3: Line 10:
Can we retitle the topic as:
"Manage display of pronouns in profile cards using the Microsoft Graph API (preview)"?

The reasons are in a [comment](https://github.com/microsoftgraph/microsoft-graph-docs/pull/19632/files#r1070223623) in pronounssettings.md.

Reviewer 3: Since I'm going to make a commit in this PR for this topic, I will include the title change to the commit.

Please let me know if you see an issue with the retitling.

Reviewer 3: Per Anton's email on 1/18 that displaying pronouns is one result of toggling on, I'm scoping down to a specific section in the concept topic where I list the exact effects of "enabling" pronouns, and for other places, like the topic and TOC titles, I try not to tie them to "display" anymore.
FYI I'm changing the topic title to "Manage pronouns settings for an organization using the Microsoft Graph API".

> If you are a global administrator, you can turn on or off the display of pronouns in your organization. By default, pronouns are **off**. If you turn pronouns on, users will be able to add and update pronouns in Microsoft 365.

Reviewer 3: @aymen-ms , this comment is to follow my email this afternoon, asking if there's any pre-existing docs for end users or administrators about the experience and support for pronouns in Microsoft 365 - normally, our developer docs do not assume that all customers are completely familiar with existing end user or admin experience related to that feature, if that applies. In the case of pronouns support, that definitely applies. 
If there's such pre-existing Microsoft docs for pronouns, please include a link. Otherwise, I'm assuming there isn't (as I have searched). In that case, this concept topic needs to introduce the pronouns feature (part of the value of such support is indeed on the end user experience). We can't just begin by assuming customers are clear what the support is like for Microsoft services.

Reviewer 3: So in the interest of time, I am going to attempt to add a preceding paragraph to explain (a) the purpose of pronouns, (b) what the support is like for end users.
As a heads-up, I might overlook and not have the most up-to-date understanding of pronouns, so please feel free to let me know and make adjustments.

> > [!NOTE]

Reviewer 3: Lines 14-15:
For consistency, please remove this note. 

It's not because the note is incorrect. The docs have [explained](https://learn.microsoft.com/en-us/graph/versioning-and-support?view=graph-rest-beta#preview-status) and tried to consistently build the expectations for customers that is something is tagged as "preview", the implications that your note spells out apply. So we don't want to dilute that convention, and cause some customers to mistake that where the note is NOT spelled out, then even if a feature or API is just tagged as "preview", it's ok to use in production apps.

Reviewer 3: Since I'm going to make a commit in this PR for this topic, I will remove this note in the commit.

> If you are a global administrator, you have the ability to turn on or off the display of pronouns in your organization. By default, pronouns are **off**. If you turn pronouns on, users will be able to add and update pronouns in Microsoft 365.

Reviewer 2: ```suggestion
If you are a global administrator, you can turn on or off the display of pronouns in your organization. By default, pronouns are **off**. If you turn pronouns on, users will be able to add and update pronouns in Microsoft 365.
```

> APIs under the `/beta` version in Microsoft Graph are subject to change. Use of these APIs in production applications is not supported.

## Confirm your current settings

Use the [get](/graph/api/organizationsettings-list-pronounssettings?view=graph-rest-beta&preserve-view=true) operation to return the current configuration of pronouns in your organization.

In this example, pronouns are disabled: 

``` http
GET https://graph.microsoft.com/beta/organization/{organizationId}/settings/pronouns
```

If successful, the response returns a `200 OK` response code and a [**pronounsSettings**](/graph/api/resources/pronounssettings?view=graph-rest-beta&preserve-view=true) object in the response body.

``` http
HTTP/1.1 200 OK
Content-Type: application/json

{
  "isEnabledInOrganization": false
}
```
 
## Scenarios

You can either make pronouns available or unavailable in your organization.

> 

Reviewer 3: Line 40:
Per my comment in the pronounssettings.md topic, I think "available" is ambiguous. 
Since I'm going to make a commit to this topic, I'll add my rephrase. Please let me know if you see an issue with the rephrase.

``` http
PATCH https://graph.microsoft.com/beta/organization/{organizationId}/settings/pronouns
Content-Type: application/json

{
  "isEnabledInOrganization": true
}
```
If successful, the response returns a `200 OK` response code and a [**pronounsSettings**](/graph/api/resources/pronounssettings?view=graph-rest-beta&preserve-view=true) object in the response body.

``` http
HTTP/1.1 200 OK
Content-Type: application/json

{
  "isEnabledInOrganization": true
}
```

## Disabling pronouns in your organization

Use the `patch` operation to make pronouns unavailable in your organization.

``` http
PATCH https://graph.microsoft.com/beta/organization/{organizationId}/settings/pronouns
Content-Type: application/json

{
  "isEnabledInOrganization": false
}
```

If successful, the response returns a `200 OK` response code and a [**pronounsSettings**](/graph/api/resources/pronounssettings?view=graph-rest-beta&preserve-view=true) object in the response body.

``` http
HTTP/1.1 200 OK
Content-Type: application/json

{
  "isEnabledInOrganization": false
}
```

> ## Important

Reviewer 3: Lines 95-99:
FYI I'm going to move this note further up in the topic in my upcoming commit.
In general, important information (like you characterized it) should not be all the way at the end of a topic. Customers mostly skim and may miss this info.

> When you turn pronouns on or off, it can take up to six hours for users to see changes. For example, if you turn pronouns on, users won’t be able to see the option to add pronouns on their profile for up to six hours. If you turn pronouns off, any previously set pronouns might stay visible in Microsoft 365 (for example, on profile cards) for up to six hours. 

Reviewer 3: Line 97, 99 - please use present tense even when you're describing a sequence of events.

FYI in general, we refrain from using future tense in documentation. After all, the behavior (involved in a sequence of events) is already implemented and is a fact. From that perspective, docs that describe current behaviors of a feature should simply use present tense.

When you take into account localization, future tense can create ambiguity.

Reviewer 3: Since I'm going to make a commit to this topic, I'll fix the tense.
